src/masternodes/rpc_poolpair.cpp

```diff
@@ -134,6 +134,13 @@ UniValue poolPathsToJSON(std::vector<std::vector<DCT_ID>> & poolPaths) {
 void CheckAndFillPoolSwapMessage(const JSONRPCRequest& request, CPoolSwapMessage &poolSwapMsg) {
     std::string tokenFrom, tokenTo;
     UniValue metadataObj = request.params[0].get_obj();
+
+    if (metadataObj["tokenFrom"].getValStr() == "") {
```

Comment:
`metadataObj["tokenFrom"].empty()`

Reply:
I did try this, but that throws the error when `tokenFrom` is not empty.

Reply:
`if (metadataObj["tokenFrom"].getValStr().empty())`